webapp/src/components/admin_settings/jitsi_settings.tsx

```diff
@@ -0,0 +1,243 @@
+
+    const onJitsiCompatibilityChanged = useCallback((e: ChangeEvent<InputElementType>) => {
+        updateSettingsState('jitsicompatibilitymode', e.target.value === 'true');
+    }, [updateSettingsState]);
+
+    const onJitsiEmbeddedChanged = useCallback((e: ChangeEvent<InputElementType>) => {
+        updateSettingsState('jitsiembedded', e.target.value === 'true');
+    }, [updateSettingsState]);
+
+    const onJitsiAuthChanged = useCallback((e: ChangeEvent<InputElementType>) => {
+        updateSettingsState('jitsijwt', e.target.value === 'true');
+    }, [updateSettingsState]);
+
+    const onJitsiMeetingLinkExpChanged = useCallback((e: ChangeEvent<InputElementType>) => {
+        updateSettingsState('jitsilinkvalidtime', e.target.value);
+    }, [updateSettingsState]);
+
+    const onJitsiMeetingNamesChanged = useCallback((e: ChangeEvent<InputElementType>) => {
+        updateSettingsState('jitsinamingscheme', e.target.value);
+    }, [updateSettingsState]);
+
+    const onJitsiURLChanged = useCallback((e: ChangeEvent<InputElementType>) => {
+        updateSettingsState('jitsiurl', e.target.value);
+    }, [updateSettingsState]);
+
+    const jitsiSection = useMemo(() => (
+        <JitsiSection
+            disabled={disabled}
+            onJitsiAppIDChange={onJitsiAppIDChanged}
+            onJitsiAppSecretChange={onJitsiAppSecretChanged}
+            onJitsiCompatibilityChange={onJitsiCompatibilityChanged}
+            onJitsiEmbeddedChange={onJitsiEmbeddedChanged}
+            onJitsiJwtAuthChange={onJitsiAuthChanged}
+            onJitsiMeetingLinkExpChange={onJitsiMeetingLinkExpChanged}
+            onJitsiMeetingNamesChange={onJitsiMeetingNamesChanged}
+            onJitsiURLChange={onJitsiURLChanged}
+            serverUrl={settings.jitsiurl ?? ''}
+            embedded={settings.jitsiembedded ?? false}
+            namingScheme={settings.jitsinamingscheme ?? JITSI_NAMING_SCHEME.WORDS}
+            jwtEnabled={settings.jitsijwt ?? false}
+            appID={settings.jitsiappid ?? ''}
+            appSecret={settings.jitsiappsecret ?? ''}
+            meetingLinkExpire={settings.jitsilinkvalidtime ?? JITSI_LINK_VALID_TIME}
+            compatibilityMode={settings.jitsicompatibilitymode ?? false}
+        />
+    ), [disabled, settings]);
+
+    const jaasSection = useMemo(() => (
+        <JaaSSection
+            disabled={disabled}
+            onApiKeyIDChange={onJaaSApiKeyChanged}
+            onAppIDChange={onJaaSAppIDChanged}
+            onPrivateKeyChange={onJaaSPrivateKeyChanged}
+            onEmbeddedChange={onJaaSEmbeddedChanged}
+            onCompatibilityChange={onJaaSCompatibilityChange}
+            appID={settings.jaasappid ?? ''}
+            apiKey={settings.jaasapikey ?? ''}
+            privateKey={settings.jaasprivatekey ?? ''}
+            embedded={settings.jitsiembedded ?? false}
+            compatibilityMode={settings.jitsicompatibilitymode ?? false}
+        />
+    ), [settings, disabled]);
```

Comment:
Maybe we shouldn't use `useMemo` here. It's very error-prone because this callback we're giving `useMemo` requires many more dependencies than just `settings` and `disabled`. e.g. `onJaaSApiKeyChanged` etc. Please either remove the calls to `useMemo`, or fill in all of the necessary deps here in the array. Same with other uses of `useMemo` in the admin console files.

Reply:
I propose we remove all of the `useCallback` and `useMemo` functionality in the admin console components, to avoid any error-prone gotchas with dep arrays etc. The performance gains are not worth it IMO, since rendering this settings page is not going to happen often, and is does not need to be particular performant with other pieces of the application. It also would make it easier to read because there's lots of indentation for each operation atm.